src/zc/relation/tokens.txt

```diff
@@ -78,18 +78,28 @@ object, similar to an intid utility. [#faux_intid]_
     >>> class Reference(object): # see zope.app.keyreference
     ...     def __init__(self, obj):
     ...         self.object = obj
-    ...     def __cmp__(self, other):
+    ...     def _get_sorting_key(self):
     ...         # this doesn't work during conflict resolution.  See
     ...         # zope.app.keyreference.persistent, 3.5 release, for current
     ...         # best practice.
-    ...         if not isinstance(other, Reference):
-    ...             raise ValueError('can only compare with Reference objects')
-    ...         if self.object._p_jar is None or other.object._p_jar is None:
+    ...         if self.object._p_jar is None:
     ...             raise ValueError(
     ...                 'can only compare when both objects have connections')
-    ...         _self = (self.object._p_jar.db().database_name,  self.object._p_oid)
-    ...         _other = (other.object._p_jar.db().database_name, other.object._p_oid)
-    ...         return (_self > _other) - (_self < _other)
+    ...         return self.object._p_oid or ''
+    ...     def __lt__(self, other):
+    ...         # this doesn't work during conflict resolution.  See
+    ...         # zope.app.keyreference.persistent, 3.5 release, for current
+    ...         # best practice.
+    ...         if not isinstance(other, Reference):
+    ...             raise ValueError('can only compare with Reference objects')
+    ...         return self._get_sorting_key() < other._get_sorting_key()
+    ...     def __eq__(self, other):
+    ...         # this doesn't work during conflict resolution.  See
+    ...         # zope.app.keyreference.persistent, 3.5 release, for current
+    ...         # best practice.
+    ...         if not isinstance(other, Reference):
+    ...             raise ValueError('can only compare with Reference objects')
+    ...         return self._get_sorting_key() == other._get_sorting_key()
```

Comment:
You want to decorate the class with [@functools.total_ordering](https://docs.python.org/3/library/functools.html#functools.total_ordering) if you're not going to define `__gt__`, `__ne__` and the rest manually.

Reply:
I did that but nothing was changing in the test results so I removed the @total_ordering decorator. I can put it again if needed

Reply:
The `@total_ordering` is *not* optional. It ensures that other ordering methods are autogenerated based on `__lt__` and `__eq__`. Can you please put it back? :-)